packages/shared-lib/src/package-manager/package.ts

```diff
@@ -172,6 +177,78 @@ export namespace ExpectedPackage {
 			}
 		}[]
 	}
+	export interface ExpectedPackageHtmlTemplate extends Base {
+		type: PackageType.HTML_TEMPLATE
+		content: {
+			/** path to the HTML template */
+			path: string
+			/** Add prefix to output artifacts */
+			outputPrefix: string
+		}
+		version: {
+			renderer?: {
+				/** Renderer width, defaults to 1920 */
+				width?: number
+				/** Renderer width, defaults to 1080 */
+				height?: number
+				/** Zoom level, defaults to 1 */
+				zoom?: number
+				/** (defaults to black) */
+				backgroundColor?: string
+				userAgent?: string
+			}
+
+			/**
+			 * Convenience settings for a template that follows the typical CasparCG steps;
+			 * update(data); play(); stop();
+			 * If this is set, steps are overridden */
+			casparCG?: {
+				/**
+				 * Data to send into the update() function of a CasparCG Template.
+				 * Strings will be piped through as-is, objects will be JSON.stringified.
+				 */
+				data: { [key: string]: any } | null | string
+
+				/** How long to wait between each action in a CasparCG template, (default: 1000ms) */
+				delay?: number
+			}
+
+			steps?: (
+				| { do: 'waitForLoad' }
+				| { do: 'sleep'; duration: number }
+				| {
+						do: 'sendHTTPCommand'
+						url: string
+						/** GET, POST, PUT etc.. */
+						method: string
+						body?: ArrayBuffer | ArrayBufferView | NodeJS.ReadableStream | string | URLSearchParams
+
+						headers?: Record<string, string>
+				  }
+				| { do: 'takeScreenshot'; fileName: string }
+				| { do: 'startRecording'; fileName: string }
+				| { do: 'stopRecording' }
+				| { do: 'cropRecording'; fileName: string }
+				| { do: 'executeJs'; js: string }
+				// Store an object in memory
+				| { do: 'storeObject'; key: string; value: Record<string, any> }
+				// Modify an object in memory. Path is a dot-separated string
+				| { do: 'modifyObject'; key: string; path: string; value: any }
+				// Send an object to the renderer as a postMessage
+				| { do: 'injectObject'; key: string }
```

Comment:
I find the naming here a bit confusing - these actions are not related - `storeObject` and `injectObject`? Since there is no `retrieveObject`, maybe it would make more sense to call it `setObject` or even `setValue` if there's no reason to limit this to objects? and then change `injectObject` into `postMessage`, since that's what's going to happen?

Also I suggest not saying "memory" and instead say "page context".

```suggestion
				// Store an object in memory
				| { do: 'setValue'; key: string; value: any }
				// Modify an object in memory. Path is a dot-separated string
				| { do: 'modifyObject'; key: string; path: string; value: any }
				// Send an object to the renderer as a postMessage
				| { do: 'postMessage'; value: string }
```